runtimes/common/src/dot_names/mod.rs

```diff
@@ -0,0 +1,74 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::ensure;
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_core::{ConstU32, RuntimeDebug};
+use sp_runtime::{BoundedVec, SaturatedConversion};
+use sp_std::vec::Vec;
+
+use pallet_web3_names::{Config, Error};
+
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum DotNameValidationError {
+	TooShort,
+	TooLong,
+	InvalidCharacter,
+}
+
+impl<T, I> From<DotNameValidationError> for Error<T, I>
+where
+	T: Config<I>,
+	I: 'static,
+{
+	fn from(value: DotNameValidationError) -> Self {
+		match value {
+			DotNameValidationError::TooLong => Self::TooLong,
+			DotNameValidationError::TooShort => Self::TooShort,
+			DotNameValidationError::InvalidCharacter => Self::InvalidCharacter,
+		}
+	}
+}
+
+#[derive(Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen, PartialEq, Eq, PartialOrd, Ord, Clone)]
+pub struct DotName<const MIN_LENGTH: u32, const MAX_LENGTH: u32>(BoundedVec<u8, ConstU32<MAX_LENGTH>>);
```

Comment:
All the changes were made to allow us to do this 😄 instead of the mess we currently have for the `AsciiWeb3Name` type we use for our own web3names.